subsys/net/lib/dns/mdns_responder.c

```diff
@@ -274,6 +277,96 @@ static int send_response(struct net_context *ctx,
 	return ret;
 }
 
+static const char *qtype_to_string(int qtype)
+{
+	switch (qtype) {
+	case DNS_RR_TYPE_A: return "A";
+	case DNS_RR_TYPE_CNAME: return "CNAME";
+	case DNS_RR_TYPE_PTR: return "PTR";
+	case DNS_RR_TYPE_TXT: return "TXT";
+	case DNS_RR_TYPE_AAAA: return "AAAA";
+	case DNS_RR_TYPE_SRV: return "SRV";
+	default: return "(unsupported)";
+	}
+}
+
+
+#if defined(CONFIG_MDNS_RESPONDER_DNS_SD)
+static void send_sd_response(struct net_context *ctx,
+		 struct net_pkt *pkt, union net_ip_header *ip_hdr,
+		 struct dns_msg_t *dns_msg, struct net_buf *result)
+{
+	int ret;
+	const struct dns_sd_rec *record;
+	struct dns_sd_rec filter;
+	struct sockaddr dst;
+	const struct in6_addr *addr6 = NULL;
+	const struct in_addr *addr4 = NULL;
+	char service_buf[DNS_SD_SERVICE_MAX_SIZE + 1];
+	char proto_buf[DNS_SD_PROTO_SIZE + 1];
+	char domain_buf[DNS_SD_DOMAIN_MAX_SIZE + 1];
+
+	/* This actually is used but the compiler doesn't see that */
+	ARG_UNUSED(record);
+
+#if defined(CONFIG_NET_IPV4)
```

Comment:
Better would be to have if (IS_ENABLED(CONFIG_NET_IPV4)) { so that the compiler will check the code.